packages/react-components/react-theme/stories/Theme/colors/ThemeColors.stories.tsx

```diff
@@ -19,125 +24,79 @@ const theme = {
-  Platinum: 'colorPalettePlatinum',
-  Anchor: 'colorPaletteAnchor',
-} as const;
-
-type GlobalSharedColors = keyof typeof colorPalette;
-
-const buttonStyle = ({ active }: { active: boolean }): React.CSSProperties => ({
-  position: 'relative',
-  verticalAlign: 'middle',
-  padding: 0,
-  margin: 0,
-  width: 40,
-  height: 40,
-  border: 'none',
-  boxShadow: active ? '0 0 0 1px white, 0 0 0 2px black' : 'none',
-  borderRadius: 0,
-  outline: 'none ',
-  zIndex: active ? 2 : 1,
+const useStyles = makeStyles({
+  searchContainer: {
+    display: 'flex',
+    alignItems: 'center',
+  },
+  inputSearch: {
+    width: '100%',
+  },
 });
 
-const ColorButton: React.FunctionComponent<
-  {
-    color: 'neutral' | GlobalSharedColors;
-    active: boolean;
-    setPreviewColor: (color: 'neutral' | GlobalSharedColors | null) => void;
-    setColor: (color: 'neutral' | GlobalSharedColors) => void;
-  } & React.ButtonHTMLAttributes<HTMLButtonElement>
-> = ({ style = {}, color, active, setPreviewColor, setColor, ...rest }) => (
-  <button
-    style={{
-      ...style,
-      ...buttonStyle({ active }),
-    }}
-    onClick={() => {
-      setColor(color);
-      setPreviewColor(null);
-    }}
-    onMouseEnter={() => setPreviewColor(color)}
-    onMouseLeave={() => setPreviewColor(null)}
-    {...rest}
-  />
+const tokens: Array<keyof Theme> = (Object.keys(theme.webLight) as Array<keyof Theme>).filter(
+  tokenName => tokenName.match(/^color(?!Palette).*/) || tokenName.startsWith(`colorPalette`),
 );
 
-const neutralTokens = (Object.keys(theme.webLight) as Array<keyof Theme>).filter(tokenName =>
-  tokenName.match(/^color(?!Palette).*/),
-);
+// It goes through all the existing tokens and returns the tokens matching the input value
+const searchToken = (inputValue: string) => {
+  const tokensFoundBySearch = tokens.filter(
+    token =>
+      token.toLowerCase().includes(inputValue) ||
+      theme.webLight[token].toString().includes(inputValue) ||
+      theme.webDark[token].toString().includes(inputValue) ||
+      theme.teamsLight[token].toString().includes(inputValue) ||
+      theme.teamsDark[token].toString().includes(inputValue) ||
+      theme.teamsHighContrast[token].toString().includes(inputValue),
+  );
+  return tokensFoundBySearch;
+};
 
 export const Colors = () => {
-  const [color, setColor] = React.useState<'neutral' | GlobalSharedColors>('neutral');
-  const [previewColor, setPreviewColor] = React.useState<'neutral' | GlobalSharedColors | null>(null);
-  const activeColor = previewColor || color;
+  const [tokensSearchResult, setTokensSearchResult] = React.useState<Array<keyof Theme>>(tokens);
+
+  // Text typed in the input bar
+  const [inputValue, setInputValue] = React.useState('');
+
+  // Value checked from the filter menu button
+  const [checkedValue, setCheckedValue] = React.useState<Record<string, string[]>>();
+
+  const styles = useStyles();
+
+  const onInputChange: InputProps['onChange'] = (_, data) => {
```

Comment:
nit: I would debounce this, I am experiencing some lag when I'm typing into the search input